Tools/WebKitTestRunner/TestController.cpp

```diff
@@ -1667,7 +1681,12 @@ bool TestController::runTest(const char* inputLine)
     
     TestOptions options = testOptionsForTest(command);
 
-    m_currentInvocation = makeUnique<TestInvocation>(adoptWK(createTestURL(command.pathOrURL.c_str())).get(), options);
+    WKURLRef testURLRef = createTestURL(command.pathOrURL.c_str());
+    if (!testURLRef)
+        return false;
+
+    m_mainResourceURL = adoptWK(testURLRef);
```

Comment:
Usually it's preferable to adopt the object when it's created.  If `createTestURL` returns `nullptr`, the `WKURLRef` will just be empty, but the check on the next line still works:

```suggestion
    auto testURLRef = adoptWK(createTestURL(command.pathOrURL.c_str()));
    if (!testURLRef)
        return false;

    m_mainResourceURL = testURLRef;
```

Also, if there's no reason not to always set `m_mainResourceURL` (even if the value is `nullptr`), then this becomes:

```suggestion
    m_mainResourceURL = adoptWK(createTestURL(command.pathOrURL.c_str()));
    if (!m_mainResourceURL)
        return false;
```